File: pbl6_be/Domain/Models/Schemas/WorkLog.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;
using Domain.Models;
using pbl6_be.Domain.Models;

namespace pbl6_be.Domain.Models.Schemas
{
    [Table("work_log")]
    public class WorkLog : TableWithIntId
    {
        [Column("check_in_time")]
        public DateTime CheckInTime { get; set; } = DateTime.MinValue;

        [Column("check_out_time")]
        public DateTime CheckOutTime { get; set; } = DateTime.MaxValue;

        [Column("mac_address")]
        [StringLength(50)]
        public string MacAddress { get; set; } = null!;

        [Column("total_time")]
        public double TotalTime { get; set; }

        [ForeignKey("user")]
        [Column("user_id", Order = 2)]
        public int UserId { get; set; }
        
        [ForeignKey("shift")]
        [Column("shift_id", Order = 3)]
        public int ShiftId { get; set; }

        public virtual User User { get; set; }
        public virtual Shift Shift { get; set; }
    }
}

File: pbl6_app/lib/model/WorkLog.dart
class WorkLog {
  final int id;
  final int userId;
  final int shiftId;
  final double totalTime;
  final DateTime checkInTime;
  final DateTime checkOutTime;

  WorkLog({
    required this.id,
    required this.userId,
    required this.shiftId,
    required this.totalTime,
    required this.checkInTime,
    required this.checkOutTime,
  });

  factory WorkLog.fromJson(Map<String, dynamic> data) {
    return WorkLog(
      id: data['id'],
      userId: data['userId'],
      shiftId: data['shiftId'],
      totalTime: data['totalTime'] as double,
      checkInTime: DateTime.parse(data['checkInTime']),
      checkOutTime: DateTime.parse(data['checkOutTime']),
    );
  }
}
